app/components/bookings/edit/edit-label.component.ts

```diff
@@ -0,0 +1,95 @@
+import {Component, Input} from '@angular/core';
+import {IONIC_DIRECTIVES, NavController, Toast} from "ionic-angular/index";
+import {Booking} from "../../../models/booking/booking";
+import {HoursToWorkdaysPipe} from "../../../pipes/hours-to-workdays.pipe";
+import {BookingService} from "../../../services/bookings/booking.service";
+
+@Component({
+  selector: 'edit-label',
+  directives: [IONIC_DIRECTIVES],
+  templateUrl: 'build/components/bookings/edit/edit-label.component.html',
+  pipes: [HoursToWorkdaysPipe]
+})
+
+export class EditLabelComponent {
+
+  private _isEditMode: boolean = false;
+  private _rememberValue: String= "";
+
+  @Input()
+  public text: String;
+  @Input()
+  public booking: Booking;
+
+  constructor( private _navController: NavController,
+               private _bookingService: BookingService) {
+  }
+
+  private _focus():void{
+    // this Method is just here for the case that this Editlabel should be triggered
+    // by a click instead of a longclick in the future
+  }
+
+  private _updateBooking ():void{
```

Comment:
You're wasting a wonderful opportunity here, to make this a really valuable generic component which could used pretty much anywhere, where a editable input field is needed, by doing the updateBooking inside the edit label component. This way it's limited to be used only for this purpose!
This would be so much cooler, if you'd just have an @Output() public onUpdate: EventEmitter<string>; property, which is emitted on update instead of calling onUpdate() and then whenever you use this component, in another component, you just assign the method you want to be called on update, in this individual context.
That way you could implement _updateBookingDescription($event) in the BookingComponent and do the actually updating there. And suddenly the EditLabelComponent becomes this awesome generic thing which could be used anywhere! 😄👍